modules/codegen/test/src/smithy4s/codegen/internals/RendererConfigSpec.scala

```diff
@@ -0,0 +1,225 @@
+      "val error: UnionSchema[HelloError] = HelloError.schema"
+    )
+    assertContainsSection(serviceCode, "def liftError(throwable: Throwable)")(
+      """|def liftError(throwable: Throwable): Option[HelloError] = throwable match {
+         |  case e: HelloError => Some(e)
+         |  case _ => None
+         |}""".stripMargin
+    )
+    assertContainsSection(serviceCode, "def unliftError(e: HelloError)")(
+      "def unliftError(e: HelloError): Throwable = e"
+    )
+
+    assertContainsSection(serviceCode, "object HelloError")(
+      """|object HelloError {
+         |  val id: ShapeId = ShapeId("hello.world", "HelloError")
+         |  val hints: Hints = Hints.empty
+         |  val schema: UnionSchema[HelloError] = {
+         |    val badRequestAlt = BadRequest.schema.oneOf[HelloError]("BadRequest")
+         |    val internalServerErrorAlt = InternalServerError.schema.oneOf[HelloError]("InternalServerError")
+         |    union(badRequestAlt, internalServerErrorAlt) {
+         |      case c: BadRequest => badRequestAlt(c)
+         |      case c: InternalServerError => internalServerErrorAlt(c)
+         |    }
+         |  }
+         |}""".stripMargin
+    )
+
+  }
+
+  private def findServiceCode(contents: List[String]): String = {
+    contents.find(_.contains("trait HelloWorldServiceGen")) match {
+      case None =>
+        fail("No generated scala file contains valid service definition")
+      case Some(svc) => svc
+    }
+  }
+
+  private def testErrorsAsUnionsDisabled(smithy: String) = {
+    val contents = generateScalaCode(smithy)
+    val serviceCode = findServiceCode(contents)
+
+    assertContainsSection(serviceCode, "override val errorable")(
+      "override val errorable: Option[Errorable[HelloError]] = Some(this)"
+    )
+    assertContainsSection(serviceCode, "val error: UnionSchema[HelloError]")(
+      "val error: UnionSchema[HelloError] = HelloError.schema"
+    )
+    assertContainsSection(serviceCode, "def liftError(throwable: Throwable)")(
+      """|def liftError(throwable: Throwable): Option[HelloError] = throwable match {
+         |  case e: BadRequest => Some(HelloError.BadRequestCase(e))
+         |  case e: InternalServerError => Some(HelloError.InternalServerErrorCase(e))
+         |  case _ => None
+         |}""".stripMargin
+    )
+    assertContainsSection(serviceCode, "def unliftError(e: HelloError)")(
+      """|def unliftError(e: HelloError): Throwable = e match {
+         |  case HelloError.BadRequestCase(e) => e
+         |  case HelloError.InternalServerErrorCase(e) => e
+         |}""".stripMargin
+    )
+
+    assertContainsSection(serviceCode, "sealed trait HelloError")(
+      """|sealed trait HelloError extends scala.Product with scala.Serializable {
+         |  @inline final def widen: HelloError = this
+         |}""".stripMargin
+    )
+
+    assertContainsSection(serviceCode, "object HelloError")(
```

Comment:
I'm just a tiny bit worried these tests may be a bit painful to maintain when an unrelated change is made...

Reply:
Sure, open to alternatives how to test it, wasn't sure it's the right way anyway. 
It's basically like `fileContent.contains(expectedSection)" assert but with nicer error messages in case of failures.